client/src/views/viewers/AudioViewer.vue

```diff
@@ -1,26 +1,80 @@
 <!-- Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS -->
 
 <template>
-  <audio
-    controls
-    v-if="src.length"
-    :src="src"
-  />
+  <file-viewer-wrapper>
+    <template #viewer>
+      <audio
+        controls
+        v-if="src.length"
+        ref="audioElement"
+        @play="updateMediaData"
+        @playing="updateMediaData"
+        @canplay="updateMediaData"
+        @pause="updateMediaData"
+        @volumechange="updateMediaData"
+        :src="src"
+      />
+    </template>
+    <!-- <template #bottombar>
+      <file-viewer-action-bar
+        :actions="[
+          { icon: paused ? play : pause, handler: togglePlayback },
+          { icon: getVolumeIcon(), handler: toggleVolume },
+        ]"
+      />
+    </template> -->
+  </file-viewer-wrapper>
 </template>
 
 <script setup lang="ts">
+import { play, pause, volumeHigh, volumeLow, volumeMedium, volumeMute } from 'ionicons/icons';
 import { onMounted, ref } from 'vue';
+import { FileViewerActionBar } from '@/components/viewers';
+import { FileViewerWrapper } from '@/views/viewers';
 import { FileContentInfo } from '@/views/viewers/utils';
 
 const props = defineProps<{
   contentInfo: FileContentInfo;
 }>();
 
+const VOLUME_LEVELS = [0, 0.25, 0.5, 1];
+
 const src = ref('');
+const audioElement = ref();
+const paused = ref(true);
```

Comment:
Do you need this? You can access the `paused` value on the element itself to get its state. Same with volume.

Reply:
Yes we need this part: audioElement as ref don't trigger its "change" state when using media controls.